lib/thumbnex.ex

```diff
@@ -76,23 +89,36 @@ defmodule Thumbnex do
     fps = number_opt(opts, :fps, 1)
     optimize = Keyword.get(opts, :optimize, true)
 
-    multi_frame_path = ExtractFrame.multiple_frames(input_path, frame_count, fps, output_ext: ".gif")
-
-    multi_frame_path
-    |> Mogrify.open
-    |> Mogrify.verbose
-    |> resize_if_different(desired_width, desired_height)
-    |> Mogrify.resize_to_limit("#{max_width}x#{max_height}")
-    |> optimize_mogrify_image(optimize)
-    |> Mogrify.save(path: output_path)
-
-    :ok = File.rm! multi_frame_path
+    ExtractFrame.multiple_frames(input_path, frame_count, fps, output_ext: ".gif")
+    |> case do
+      {:ok, multi_frame_path} ->
+        multi_frame_path
+        |> Mogrify.open()
+        |> Mogrify.verbose()
+        |> resize_if_different(desired_width, desired_height)
+        |> Mogrify.resize_to_limit("#{max_width}x#{max_height}")
+        |> optimize_mogrify_image(optimize)
+        |> Mogrify.save(path: output_path)
+
+        File.rm(multi_frame_path)
+        |> case do
+          :ok ->
+            {:ok, nil}
```

Comment:
Same here, should probably return `:ok` (also it no longer matches the `@spec` if returning `{:ok, nil}`)

Reply:
fixed